### DISCUSSION TITLE
可以请问参考文献要怎么配置才能显示出来吗

### DISCUSSION THREAD

**Discussion by devindang** [note286/xduts#150] [Q&A] [answered]
### 步骤：
1. 设置 bib-backend 为 biblatex；
2. 设置 bib-resource；
3. `\usepackage[backend=biber]{biblatex}
\addbibresource{ref.bib`
4. 在文件内引用；
5. 使用 xelatex -> biber -> xelatex 编译。

![CleanShot 2023-10-24 at 20 48 25@2x](https://github.com/note286/xduts/assets/39517801/cca64c42-e457-4245-aa11-505808a0c487)

### 现象：

![CleanShot 2023-10-24 at 20 52 24@2x](https://github.com/note286/xduts/assets/39517801/1fede9f2-6b0f-44f3-98a7-5ae42cc762a1)

**Comment by JiaxiangHuang** [comment]
昨天我的也出不来
我用的 texstudio使用bibtex,
打开yourthiese.tex 按F5编译.tex
打开yourthiese.aux 按F8编译bibtex
打开yourthiese.tex 按F5编译.tex
打开yourthiese.tex 按F5编译.tex
能出来

**Reply by devindang** [reply]
后面要xelatex编译两次吗？我看linter提示的是biber 编译一次，xelatex编译一次

**Comment by note286** [comment, marked as answer]
`biblatex`和`bibltex`二选一即可。

# biblatex

手册里写的很清楚，默认为`biblatex`。

![image](https://github.com/note286/xduts/assets/16207292/552f336e-31e9-4087-9f23-4aa064666e4f)


```latex
\documentclass{xdupgthesis}
\xdusetup{info/bib-resource={biblatex-examples.bib}}
\begin{document}
\chapter{title}
\cite{angenendt}\cite{augustine}
\end{document}
```

```
xelatex -synctex=-1 main
biber main
xelatex -synctex=-1 main
```

![image](https://github.com/note286/xduts/assets/16207292/3ef92133-893b-46f6-aa10-b0232a65368b)

# bibtex

![image](https://github.com/note286/xduts/assets/16207292/9e27d236-8b50-42fe-af61-3af034bd5cdd)


```latex
\documentclass{xdupgthesis}
\xdusetup{style/bib-backend=bibtex,info/bib-resource={biblatex-examples.bib}}
\begin{document}
\chapter{title}
\cite{angenendt}\cite{augustine}
\end{document}
```

```
xelatex -synctex=-1 main
bibtex main
xelatex -synctex=-1 main
xelatex -synctex=-1 main
```

![image](https://github.com/note286/xduts/assets/16207292/1017f065-7119-42e6-9efe-db0b30d1ed72)

**Reply by devindang** [reply]
非常感谢，请问是不是使用了 noauxwrite ，在参考文献处就不会显示。我刚接触 latex，没有在 google 上检索到关于这个的信息

**Reply by note286** [reply]
如果你不了解，你可以当xduts提供的`\noauxwrite`命令不存在。

**Reply by devindang** [reply]
好的，谢谢！🙏

**Reply by XCJ-LateFall** [reply]
你好，我在texstudio中使用和你一样的方式编译tex文件，参考文献并未在论文末尾列出，在文献引用处也并无[1][2]出现，而是显示rahaman2007reviewrahaman2007review，请问问题出在哪里？

tex文件内容如下：
\documentclass{xdupgthesis}
\xdusetup{info/bib-resource={bibresourece.bib}}
\begin{document}
	\chapter{title}
	\cite{rahaman2007review}\cite{rahaman2007review}
\end{document}

编译顺序如下：
F5 xelatex → F8 biber → F5 xelatex

出现的相关的警告如下：
Citation 'rahaman2007review' on page 1 undefined
Empty bibliography
There were undefined references.
Please (re)run Biber on the file:(biblatex) test(biblatex) and rerun LaTeX afterwards.

**Reply by note286** [reply]
@XCJ-LateFall 建议你先尝试使用命令编译，如果没有问题，那么就是TeXstudio配置不正确或者使用不正确。

关于TeXstudio的使用，你可以参考 https://github.com/note286/xduts/discussions/40#discussioncomment-4956807 下面的评论，可能对你有所帮助。

**Reply by XCJ-LateFall** [reply]
运行biber.exe后出现以下报错
开始 : biber.exe "test"

INFO - This is Biber 2.19
INFO - Logfile is 'test.blg'
INFO - Reading 'test.bcf'
INFO - Found 1 citekeys in bib section 0
INFO - Processing section 0
INFO - Looking for bibtex file 'bibresourece.bib' for section 0

ERROR - Cannot find 'bibresourece.bib'!
INFO - ERRORS: 1

出现错误

**Reply by note286** [reply]
> ERROR - Cannot find 'bibresourece.bib'!
